packages/world/gas-report.txt

```diff
@@ -1,26 +1,26 @@
-(test/UniqueEntityModule.t.sol) | install unique entity module [world.installModule(uniqueEntityModule, new bytes(0))]: 773726
-(test/UniqueEntityModule.t.sol) | get a unique entity nonce (non-root module) [uint256 uniqueEntity = uint256(getUniqueEntity(world))]: 65723
-(test/UniqueEntityModule.t.sol) | installRoot unique entity module [world.installRootModule(uniqueEntityModule, new bytes(0))]: 780192
-(test/UniqueEntityModule.t.sol) | get a unique entity nonce (root module) [uint256 uniqueEntity = uint256(getUniqueEntity(world))]: 65723
-(test/World.t.sol) | Delete record [world.deleteRecord(namespace, file, singletonKey)]: 16137
-(test/World.t.sol) | Push data to the table [world.pushToField(namespace, file, keyTuple, 0, encodedData)]: 96529
+(test/KeysWithValueModule.t.sol) | install keys with value module [world.installRootModule(keysWithValueModule, abi.encode(sourceTableId))]: 600366
+(test/KeysWithValueModule.t.sol) | set a record on a table with KeysWithValueModule installed [world.setRecord(namespace, sourceFile, keyTuple1, abi.encodePacked(value))]: 191672
+(test/KeysWithValueModule.t.sol) | install keys with value module [world.installRootModule(keysWithValueModule, abi.encode(sourceTableId))]: 600366
+(test/KeysWithValueModule.t.sol) | change a record on a table with KeysWithValueModule installed [world.setRecord(namespace, sourceFile, keyTuple1, abi.encodePacked(value2))]: 149170
+(test/KeysWithValueModule.t.sol) | delete a record on a table with KeysWithValueModule installed [world.deleteRecord(namespace, sourceFile, keyTuple1)]: 57914
+(test/KeysWithValueModule.t.sol) | install keys with value module [world.installRootModule(keysWithValueModule, abi.encode(sourceTableId))]: 600366
+(test/KeysWithValueModule.t.sol) | set a field on a table with KeysWithValueModule installed [world.setField(namespace, sourceFile, keyTuple1, 0, abi.encodePacked(value1))]: 199906
+(test/KeysWithValueModule.t.sol) | change a field on a table with KeysWithValueModule installed [world.setField(namespace, sourceFile, keyTuple1, 0, abi.encodePacked(value2))]: 155751
+(test/UniqueEntityModule.t.sol) | install unique entity module [world.installModule(uniqueEntityModule, new bytes(0))]: 773660
+(test/UniqueEntityModule.t.sol) | get a unique entity nonce (non-root module) [uint256 uniqueEntity = uint256(getUniqueEntity(world))]: 65679
+(test/UniqueEntityModule.t.sol) | installRoot unique entity module [world.installRootModule(uniqueEntityModule, new bytes(0))]: 780148
+(test/UniqueEntityModule.t.sol) | get a unique entity nonce (root module) [uint256 uniqueEntity = uint256(getUniqueEntity(world))]: 65679
+(test/World.t.sol) | Delete record [world.deleteRecord(namespace, file, singletonKey)]: 16115
+(test/World.t.sol) | Push data to the table [world.pushToField(namespace, file, keyTuple, 0, encodedData)]: 119235
```

Comment:
Around 20k gas increase here 😢 I'm not sure how much data was pushed here though (with the additional delegatecall indirection we also pay for calldata for calling the DynamicPartial method). Also it seems like we're missing a gas report for `updateInField`.

Reply:
Agree this PR is necessary to reduce the World size. How many more DynamicPartial method do you think we'll need? If there are other less used methods like `registerStoreHook` that we could externalize that might be a good alternative too.

I'm down to merge as is though to unblock you and then try to reduce gas overhead in a followup by reducing the general overhead of calling dynamic function selectors and by moving frequently called methods to the World internal code and externalizing less often used methods instead.